drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1681,7 +1681,7 @@ static void uarte_nrfx_irq_callback_set(const struct device *dev,
 	data->int_driven->cb = cb;
 	data->int_driven->cb_data = cb_data;
 
-#if defined(CONFIG_UART_EXCLUSIVE_API_CALLBACKS)
+#if defined(UARTE_ANY_ASYNC) && defined(CONFIG_UART_EXCLUSIVE_API_CALLBACKS)
 	data->async->user_callback = NULL;
 	data->async->user_data = NULL;
 #endif
```

Comment:
A similar problem will occur for https://github.com/zephyrproject-rtos/zephyr/blob/4705482094d1f30f546f00c192c70923933612cb/drivers/serial/uart_nrfx_uarte.c#L940-L943
The first line there should be
```
 #if defined(CONFIG_UART_EXCLUSIVE_API_CALLBACKS) && defined(UARTE_INTERRUPT_DRIVEN)
```
Also `uart_nrfx_uart.c` needs a second fix - `CONFIG_UART_0_INTERRUPT_DRIVEN` needs to be checked there.

Reply:
Changed, thanks. Those did not trigger in any of the run though, may be worth looking into the tests configurations as a followup.